lisa/operating_system.py

```diff
@@ -1955,6 +1960,21 @@ def add_azure_core_repo(
             sudo=True,
         )
 
+    def create_local_repo(self, node: "RemoteNode", source_tarball: Path) -> None:
```

Comment:
It should be a private method in CBLMariner, but call it by `add_repository`. So, `add_repository` support both url and local tar files.

Reply:
updated

Reply:
Don't change the arguments of `add_repository`, so it's easy to called by any place. You can detect it's a local or remote repo in the method.

Reply:
Should I get those variables through `kwargs.pop()`? Or should I just reuse the `repo` parameter to specify a path to src tarball instead of repo url? I'd prefer the former to be more explicit of the usage.

Reply:
updated